src/common/fam_ops_libfabric.h

```diff
@@ -328,6 +331,8 @@ class Fam_Ops_Libfabric : public Fam_Ops {
     };
     std::vector<fi_addr_t> *get_fiAddrs() { return fiAddrs; };
     std::map<uint64_t, Fam_Region_Map_t *> *get_fiMrs() { return fiMrs; };
+
+    // Use context_id instead of nodeID
```

Comment:
Is this map parsed on every operation ? Is that a costly operation, Do we need to take locks?

Reply:
We can use vector instead of a map and access directly using contextId as index, similar to fiAddrs vector.